OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/NfcOperationActivity.java

```diff
@@ -67,6 +68,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
 
+        mInputParcel = getIntent().getParcelableExtra(EXTRA_CRYPTO_INPUT);
+
+        if (mInputParcel == null) {
+            // for compatibility when used from OpenPgpService
+            // (or any place other than CryptoOperationHelper)
+            mInputParcel = new CryptoInputParcel(mRequiredInput.mSignatureTime);
```

Comment:
This looks like a change that could introduce future bugs. You assume that the NFC operation is a signing operation here.